pkg/controller/baremetalasset/baremetalasset_controller_test.go

```
@@ -0,0 +1,570 @@
+			}
+		})
+	}
+}
+
+func TestCheckAssetSecret(t *testing.T) {
+	tests := []struct {
+		name         string
+		existingObjs []runtime.Object
+		expectErr    bool
+	}{
+		{
+			name:         "SecretNotFound",
+			existingObjs: []runtime.Object{},
+			expectErr:    true,
+		},
+		{
+			name:         "SecretFound",
+			existingObjs: []runtime.Object{NewSecret()},
+			expectErr:    false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fake.NewFakeClientWithScheme(scheme.Scheme, test.existingObjs...)
+			rbma := &ReconcileBareMetalAsset{
+				client: fakeClient,
+				scheme: scheme.Scheme,
+			}
+			err := rbma.checkAssetSecret(NewBMA(), log)
+			if test.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestEnsureLabels(t *testing.T) {
+	tests := []struct {
+		name         string
+		existingObjs []runtime.Object
+		expectErr    bool
+	}{
+		{
+			name:         "EnsureLabels",
+			existingObjs: []runtime.Object{NewBMA()},
+			expectErr:    false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fake.NewFakeClientWithScheme(scheme.Scheme, test.existingObjs...)
+			rbma := &ReconcileBareMetalAsset{
+				client: fakeClient,
+				scheme: scheme.Scheme,
+			}
+			err := rbma.ensureLabels(NewBMAWithClusterDeployment(), log)
+			if test.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestCheckClusterDeployment(t *testing.T) {
+	tests := []struct {
+		name         string
+		existingObjs []runtime.Object
+		expectErr    bool
+		bma          *midasv1alpha1.BareMetalAsset
+	}{
+		{
+			name:         "No cluster specified",
+			existingObjs: []runtime.Object{},
+			expectErr:    true,
+			bma:          &midasv1alpha1.BareMetalAsset{},
+		},
+		{
+			name:         "BMA With RelatedObjects",
+			existingObjs: []runtime.Object{},
+			expectErr:    true,
```

For expectErr, we should use the actual error that we expect to get back instead of a boolean.

---

`expectErr` as a boolean variable is used to test different behaviors when a test case should throw an error. David and I discussed and decided that we do not care about the exact error messages but we should only test if we get an error or not. @djzager

---

https://github.com/open-cluster-management/multicluster-inventory/pull/33#discussion_r387905248

I do believe it would be cumbersome to compare `Condition`s, but I think it is reasonable to check the kind of error you receive in these different scenarios (beyond `== nil` or `!= nil`). I wouldn't recommend you worry about the specific error string you get back, just that the error is of a particular type.

---

Done